File: cli/src/commands/list.rs
use crate::client::CowikiClient;
use crate::error::CliError;
use crate::output;

pub async fn run(
    client: &CowikiClient,
    branch: &str,
    json: bool,
) -> Result<(), CliError> {
    let pages = client.list_pages(branch).await?;

    if pages.is_empty() {
        output::print_info(&format!("no pages on branch \"{branch}\""));
        return Ok(());
    }

    if json {
        println!(
            "{}",
            serde_json::to_string_pretty(&pages)
                .map_err(|e| CliError::Unexpected(format!("failed to serialize pages: {e}")))?
        );
        return Ok(());
    }

    let mut table = comfy_table::Table::new();
    table.set_header(vec!["SLUG", "TITLE", "UPDATED"]);
    for p in &pages {
        table.add_row(vec![&p.slug, &p.title, &p.updated_at]);
    }
    output::print_table(table);

    Ok(())
}

File: cli/src/commands/list.ts
import { Command } from 'commander';
import { loadConfig } from '../config.js';
import { CowikiClient } from '../client.js';
import { printError, printInfo, printJson, printTable } from '../output.js';
import { resolveUserBranch, requireWorkspace } from '../shared.js';
import type { GlobalOpts } from '../shared.js';

export function registerListCommand(program: Command): void {
  program
    .command('list')
    .description('List wiki pages (supports --dir for wiki/entities/concepts/all)')
    .option('--dir <dir>', 'Directory: wiki (default), entities, concepts, all, or subdir like wiki/messi')
    .action(async (opts, cmd) => {
      const globalOpts = cmd.parent?.optsWithGlobals() as GlobalOpts;
      const config = loadConfig(globalOpts?.server)
      const ws = requireWorkspace(globalOpts.workspace);
      const client = new CowikiClient(config.baseUrl, config.apiKey);
      const branch = await resolveUserBranch(client);

      try {
        const pages = await client.listPages(ws, branch, opts.dir);

        if (globalOpts.json) {
          printJson(pages);
        } else if (pages.length === 0) {
          printInfo(`no pages on branch "${branch}"`);
        } else {
          printTable(
            ['SLUG', 'TITLE', 'BRANCH'],
            pages.map((p) => [p.slug, p.title, p.branch]),
          );
        }
      } catch (e) {
        printError(e instanceof Error ? e.message : String(e));
        process.exit(1);
      }
    });
}
